data/serializer/src/main/java/tech/pegasys/teku/provider/GetStateResponseV2Deserializer.java

```diff
@@ -39,9 +39,7 @@ public GetStateResponseV2 deserialize(final JsonParser jp, final Deserialization
     JsonNode node = jp.getCodec().readTree(jp);
     final Version version =
         Version.valueOf(node.findValue("version").asText().toLowerCase(Locale.ROOT));
-    final JsonNode executionOptimisticNode = node.findValue("execution_optimistic");
-    final Boolean executionOptimistic =
-        executionOptimisticNode != null ? executionOptimisticNode.asBoolean() : null;
+    final boolean executionOptimistic = node.findValue("execution_optimistic").asBoolean();
```

Comment:
Probably need to be careful on this change - theres definitely cases where execution_optimistic isn't set, so null can have meaning... 
It defaults to false, so could potentially attempt to find the value and default to false if it's not present...

Reply:
If the field isn't present, I would expect `node.findValue(...)` to throw an exception when dereferenced. It will only default to false if there's a value (any value). But are you sure there are cases where it's not set? The original code would throw a NPE in that case. PR #5503 changes it to "always include the `execution_optimistic` flag in responses."

Reply:
that'll be why it's checking null is all, no field existed, so set value to null... I'll have a closer look once i clean up my current PR 'mess' :)

Reply:
Thank you. Just to be clear, if you call `new GetStateResponseV2(version, executionOptimistic, state)` when `executionOptimistic` is null, it will crash. This is because the parameter is a primitive:

https://github.com/ConsenSys/teku/blob/e8b4ce8ef582d158b96a3f74bbefe7f62a448185/data/serializer/src/main/java/tech/pegasys/teku/api/response/v2/debug/GetStateResponseV2.java#L46-L49

Reply:
fair point... I'm not sure where it'd ever be used now, which was the other aspect, but this change certainly hasn't made anything worse... :)